frontend/src/tree/TreeItemContextMenu.tsx

```diff
@@ -44,60 +68,114 @@ export const TreeItemContextMenu = ({
     }
   };
 
+  const [deleteTreeItem, { loading: deleteTreeItemLoading, data: deleteTreeItemData, error: deleteTreeItemError }] =
+    useMutation<GQLDeleteTreeItemData, GQLDeleteTreeItemVariables>(deleteTreeItemMutation);
+
+  const deleteItem = () => {
+    const input: GQLDeleteTreeItemInput = {
+      id: uuid(),
+      editingContextId,
+      representationId: treeId,
+      treeItemId: item.id,
+    };
+    deleteTreeItem({ variables: { input } });
+    onClose;
+  };
+
+  useEffect(() => {
+    if (!deleteTreeItemLoading) {
+      if (deleteTreeItemError) {
+        setState({
+          message: 'An error has occurred while executing this action, please contact the server administrator',
+        });
+      }
+      if (deleteTreeItemData) {
+        const keys = Object.keys(deleteTreeItemData);
```

Comment:
The code should look like this:

```
const { deleteTreeItem } = deletTreeItemData;
if (isErrorPayload(deleteTreeItem)) {
  const { message } = deleteTreeItem;
  setState({ message });
} 
```

You are not using TypeScript and you are not using the message from the error payload